Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -97,10 +99,14 @@ AdvancedBusEventHandlers advancedBusEventHandlers
 
 
         /// <inheritdoc />
-        public bool IsConnected => connection.IsConnected;
+        public bool IsConnected => producerConnection.IsConnected && consumerConnection.IsConnected;
```

Comment:
That is, now in the bus we will have two physical connections instead of one? And now in the logs, all records about connecting to the broker will occur twice (if you are using publishing and consuming together)?

Reply:
Sure. I forgot to add PersistentConnectionType to the logs to be able to understand which one of connections are affected.

Reply:
Also, I'm not sure that current logic here is right.

```
        public bool IsConnected => producerConnection.IsConnected && consumerConnection.IsConnected;
```

Probably we could change it to this:

```
        public bool IsConnected => producerConnection.IsConnected || consumerConnection.IsConnected;
```

But it might be more useful to do the following:
1. Convert `IsConnected` property(or to add) to method with the following signature `bool IsConnected(PersistentConnectionType)`
2. Convert `void Connect()` to `void Connect(PersistentConnectionType)`.

Not sure what to do the best here.

Reply:
@micdenny @zidad @sungam3r What do you think?

Reply:
It also seemed to me somewhat strange, but I did not affect attention. I think `IsConnected(PersistentConnectionType)`+`Connect(PersistentConnectionType)` will be OK.

Reply:
and i will also add overloads for Connect() and IsConnected (as property as before that checks both connection)

Reply:
@Pliner ⬆️